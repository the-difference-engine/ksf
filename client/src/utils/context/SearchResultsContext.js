import React, { createContext, useState } from 'react';

export const SearchResultDataContext = createContext();

export const SearchResultDataProvider = (props) => {
  const [SearchResultData, setSearchResultData] = useState([]);

  return (
    <SearchResultDataContext.Provider
      value={[SearchResultData, setSearchResultData]}
    >
      {props.children}
    </SearchResultDataContext.Provider>
  );
};
