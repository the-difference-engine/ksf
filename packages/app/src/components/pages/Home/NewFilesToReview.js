import React, { useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import NewNomination from './NewNomination';
import styles from './styles.css';
import useSort from './useSort';

const NewFilesToReview = () => {
  const [showAll, setShowAll] = useState(false);
  const { sortedNoms, requestSort, sortConfig } = useSort();
  const sortedNominations = sortedNoms ? sortedNoms.filter((nominations) => nominations.status === 'received') : [];

  const handleClick = () => {
    setShowAll(!showAll);
  };

  const conditionalNominationRender = () => {
    if (sortedNoms && showAll) {
      return sortedNominations;
    }
    if (sortedNoms && !showAll) {
      return sortedNominations.slice(0, 3);
    }
  };

  const renderSortArrow = (columnName) => {
    return sortConfig && sortConfig.key === columnName && <FontAwesomeIcon icon={sortConfig.direction === 'ascending' ? 'arrow-down' : 'arrow-up'} />;
  };

  const renderSortableCell = (key, label) => {
    return (
      <h2 onClick={() => requestSort(key)} className="sortable-column">
        <strong>{label}</strong>
        <>{renderSortArrow(key)}</>
      </h2>
    );
  };

  return (
    <table className="new-files-table">
      <thead>
        <tr>
          <td className="add-padding-left new-files-title">
            <FontAwesomeIcon icon="file-image" color="green" />
            <h1>New Files To Review</h1>
          </td>
          <td></td>
          <td></td>
          <td></td>
          <td className="new-files-see-more">
            <div onClick={handleClick}>
              <FontAwesomeIcon icon={showAll ? 'chevron-circle-up' : 'chevron-circle-down'} />
            </div>
          </td>
        </tr>
      </thead>
      <tbody>
        <tr className="home-new-files-headers">
          <td className="add-padding-left"> {renderSortableCell('nominationName', 'Application Name')} </td>
          <td> {renderSortableCell('providerName', 'HP Name')} </td>
          <td> {renderSortableCell('representativeName', 'Family Member Name')} </td>
          <td> {renderSortableCell('dateReceived', 'Submission Date')} </td>
          <td>
            <h2>
              <strong>Stage</strong>
            </h2>
          </td>
        </tr>
        {sortedNominations ? (
          conditionalNominationRender().map((nomination) => <NewNomination nomination={nomination} key={nomination.id} />)
        ) : (
          <tr>
            <td className="add-padding-left new-files-title">
              <h1>No new nominations.</h1>
            </td>
          </tr>
        )}
      </tbody>
    </table>
  );
};

export default NewFilesToReview;
