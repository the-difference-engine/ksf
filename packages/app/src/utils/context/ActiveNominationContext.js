import React, { createContext, useEffect, useState } from 'react';

export const ActiveNominationContext = createContext();

export const ActiveNominationProvider = (props) => {
  const [activeNomination, setActiveNomination] = useState({});

  return (
    <ActiveNominationContext.Provider
      value={[activeNomination, setActiveNomination]}
    >
      {props.children}
    </ActiveNominationContext.Provider>
  );
};
