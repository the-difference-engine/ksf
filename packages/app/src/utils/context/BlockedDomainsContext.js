import React, { createContext, useEffect, useState } from 'react';

export const BlockedDomainsContext = createContext();

export const BlockedDomainsProvider = (props) => {
    const [blockedDomains, setBlockedDomains] = useState([]);

    return(
        <BlockedDomainsContext.Provider
            value={[blockedDomains, setBlockedDomains]} >
            {props.children}
        </BlockedDomainsContext.Provider>
    )
}