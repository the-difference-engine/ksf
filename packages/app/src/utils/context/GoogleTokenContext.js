import React, { createContext, useState, useEffect } from 'react';

export const GoogleTokenDataContext = createContext();

export const GoogleTokenDataProvider = (props) => {
  const [googleToken, setGoogleToken] = useState("");

  useEffect(() => {
  }, [googleToken])

  return (
    <GoogleTokenDataContext.Provider
      value={[googleToken, setGoogleToken]}
    >
      {props.children}
    </GoogleTokenDataContext.Provider>
  );
};
