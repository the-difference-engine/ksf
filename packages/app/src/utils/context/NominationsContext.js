import React, { createContext, useEffect, useState } from 'react';
import nominationsAPI from '../API/nominationsAPI';
import states from 'us-state-codes';

export const NominationsDataContext = createContext();

export const NominationsDataProvider = (props) => {
  const [NominationsData, setNominationsData] = useState();

  useEffect(() => {
    findAllNominations();
  }, []);

  const nomName = (n) => {
    const lastName = n.patientName ? n.patientName.split(' ')[1] : '';
    const state = states.getStateCodeByStateName(n.hospitalState);
    return `${lastName}-${state}`;
  };

  function findAllNominations() {
    nominationsAPI
      .getNominations()
      .then((res) => {
        const nominations = res.data;
        nominations.forEach((nomination) => {
          nomination.nominationName = nomName(nomination);
          nomination.dateReceived = new Date(
            nomination.dateReceived
          ).toLocaleDateString();
        });
        setNominationsData(nominations);
      })
      .catch((err) => console.log(err));
  }

  return (
    <NominationsDataContext.Provider
      value={[NominationsData, setNominationsData]}
    >
      {props.children}
    </NominationsDataContext.Provider>
  );
};

// export default { findAllNominations }