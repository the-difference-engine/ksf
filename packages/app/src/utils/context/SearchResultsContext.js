import React, { createContext, useState } from 'react';

export const SearchResultDataContext = createContext();

export const SearchResultDataProvider = (props) => {
  const [SearchResultData, setAllSearchResultData] = useState([
    'hello SearchResultsProvider changed path',
  ]);

  return (
    <SearchResultDataContext.Provider
      value={[SearchResultData, setAllSearchResultData]}
    >
      {props.children}
    </SearchResultDataContext.Provider>
  );
};
