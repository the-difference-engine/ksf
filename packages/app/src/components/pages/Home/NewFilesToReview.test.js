import React from "react";
import { render } from "@testing-library/react";
import NewFilesToReview from "./NewFilesToReview";
import { NominationsDataProvider } from '../../../utils/context/NominationsContext';

it("Renders new files to review component on the home page", () => {
  const getByText = render(
    <NominationsDataProvider>
      <NewFilesToReview />
    </NominationsDataProvider>
  )
    expect(getByText).toMatchSnapshot();

});
