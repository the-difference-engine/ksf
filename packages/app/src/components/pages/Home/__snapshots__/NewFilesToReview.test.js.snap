// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders new files to review component on the home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="new-files-table"
      >
        <thead>
          <tr>
            <td
              class="add-padding-left new-files-title"
            >
              <h1>
                New Files To Review
              </h1>
            </td>
            <td />
            <td />
            <td />
            <td />
            <td
              class="new-files-see-more"
            >
              <div />
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            class="home-new-files-headers"
          >
            <td
              class="add-padding-left"
            >
              <h2>
                <strong>
                  Application Name
                </strong>
              </h2>
            </td>
            <td>
              <h2>
                <strong>
                  HP Name
                </strong>
              </h2>
            </td>
            <td>
              <h2>
                <strong>
                  Family Member Name
                </strong>
              </h2>
            </td>
            <td>
              <h2>
                <strong>
                  Received Date
                </strong>
              </h2>
            </td>
            <td>
              <h2>
                <strong>
                  Stage
                </strong>
              </h2>
            </td>
            <td />
          </tr>
          <tr>
            <td>
              No new nominations.
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="new-files-table"
    >
      <thead>
        <tr>
          <td
            class="add-padding-left new-files-title"
          >
            <h1>
              New Files To Review
            </h1>
          </td>
          <td />
          <td />
          <td />
          <td />
          <td
            class="new-files-see-more"
          >
            <div />
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          class="home-new-files-headers"
        >
          <td
            class="add-padding-left"
          >
            <h2>
              <strong>
                Application Name
              </strong>
            </h2>
          </td>
          <td>
            <h2>
              <strong>
                HP Name
              </strong>
            </h2>
          </td>
          <td>
            <h2>
              <strong>
                Family Member Name
              </strong>
            </h2>
          </td>
          <td>
            <h2>
              <strong>
                Received Date
              </strong>
            </h2>
          </td>
          <td>
            <h2>
              <strong>
                Stage
              </strong>
            </h2>
          </td>
          <td />
        </tr>
        <tr>
          <td>
            No new nominations.
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
